main.py

```diff
@@ -148,143 +143,34 @@ def check_config(config):
                          "specified. Please do not specify seed.")
 
 
-def save_predictions(trainer, model, dataloader, predict_out_path):
-    batch_predictions = trainer.predict(model, dataloaders=dataloader)
-    pred_labels = np.vstack([batch['top_k_pred'] for batch in batch_predictions])
-    pred_scores = np.vstack([batch['top_k_pred_scores'] for batch in batch_predictions])
-    with open(predict_out_path, 'w') as fp:
-        for pred_label, pred_score in zip(pred_labels, pred_scores):
-            out_str = ' '.join([f'{model.classes[label]}:{score:.4}' for label, score in zip(pred_label, pred_score)])
-            fp.write(out_str+'\n')
-    logging.info(f'Saved predictions to: {predict_out_path}')
-
-
 def main():
     # Get config
     config = get_config()
     check_config(config)
-    config.run_name = '{}_{}_{}'.format(
-        config.data_name,
-        Path(config.config).stem if config.config else config.model_name,
-        datetime.now().strftime('%Y%m%d%H%M%S'),
-    )
+
     # Set up logger
     log_level = logging.WARNING if config.silent else logging.INFO
     logging.basicConfig(
         level=log_level, format='%(asctime)s %(levelname)s:%(message)s')
-    logging.info(f'Run name: {config.run_name}')
-
-    # Set up seed & device
-    set_seed(seed=config.seed)
-    device = init_device(use_cpu=config.cpu)
-
-    # Load dataset
-    datasets = data_utils.load_datasets(data_dir=config.data_dir,
-                                        train_path=config.train_path,
-                                        test_path=config.test_path,
-                                        val_path=config.val_path,
-                                        val_size=config.val_size,
-                                        is_eval=config.eval)
-
-    # Set up trainer
-    checkpoint_dir = os.path.join(config.result_dir, config.run_name)
-    checkpoint_callback = ModelCheckpoint(dirpath=checkpoint_dir,
-                                          filename='best_model',
-                                          save_last=True, save_top_k=1,
-                                          monitor=config.val_metric, mode='max')
-    earlystopping_callback = EarlyStopping(patience=config.patience,
-                                           monitor=config.val_metric, mode='max')
-    trainer = pl.Trainer(logger=False,
-                         num_sanity_val_steps=0,
-                         gpus=0 if config.cpu else 1,
-                         progress_bar_refresh_rate=0 if config.silent else 1,
-                         max_epochs=config.epochs,
-                         callbacks=[checkpoint_callback, earlystopping_callback])
 
-    # Dump config to log
-    log_path = os.path.join(checkpoint_dir, 'logs.json')
-    dump_log(log_path, config=config)
+    if not config.linear:
```

Comment:
This place can now be
if config.linear:
# load raw texts and generate tfidf, or load tfidf
else:
trainer = ...